### DISCUSSION TITLE
How do I synchronize data from the development environment database to the production environment database?

### DISCUSSION THREAD

**Discussion by evanlong-me** [prisma/prisma#20693] [Q&A] [answered]
### Question

Is there any good way?

### Information about Prisma Schema, Client Queries and Environment (optional)

```env
# .env.local
DATABASE_URL="postgresql://xxx:xxx@xxx:5432/develop"
```

```env
# .env
DATABASE_URL="postgresql://xxx:xxx@xxx:5432/production"
```

**Comment by ludralph** [comment, marked as answer]
HI @evanlong0926 👋 

In Prisma, the primary purpose of migrations is to synchronize the schema and structure of the database across different environments, such as from development to production. However, migrating data itself is not typically within the scope of Prisma's migration system. Data migration is a separate process and should be carefully planned and executed to ensure data integrity and consistency. 

Depending on the changes you've made in your development environment that need to be migrated to production, you may need to write custom scripts to perform data migration. This could involve exporting data from the development database, transforming it if necessary, and then importing it into the production database. You should, however, note that Prisma's migration system is primarily focused on schema synchronization and not data migration.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by evanlong-me** [reply]
Are there any code examples for Prisma from development to production?

I probably need a complete example to learn the optimal design scheme for Prisma.

**Reply by ludralph** [reply]
We have the [Prisma examples repository](https://github.com/prisma/prisma-examples) that shows how to use Prisma with different use cases. There is also an [example](https://vercel.com/guides/nextjs-prisma-postgres) that shows how to use Prisma with Next.js and Vercel Postgres and deploy to Vercel.
